sympy/stats/crv_types.py

```diff
@@ -143,6 +143,15 @@ def rv(symbol, cls, args):
     args = list(map(sympify, args))
     dist = cls(*args)
     dist.check(*args)
+    if any([isinstance(arg, RandomSymbol) for arg in args]):
+        symbol = sympify(symbol)
+        new_dist = find_dist(cls, args)
+        if  new_dist != None:
+            return rv(symbol, new_dist[0], new_dist[1])
+        _pdf = dist.pdf(symbol)
+        _pdf, _set = dist.handle_compound_dist(_pdf), dist.set
```

Comment:
I would not calculate the integral at this level. I would recommend creating a marginalized joint distribution, without performing the integration.